internal/pkg/db/redis/client.go

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright 2018 Redis Labs Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package redis
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+	"github.com/gomodule/redigo/redis"
+)
+
+type connectionType int
+
+const (
+	connTypeTCP connectionType = 1 << iota
+	connTypeUDS
+	connTypeEredis
+)
+
+const (
+	protoTCP    = "tcp"
+	protoUDS    = "unix"
+	protoEredis = "eredis"
+)
+
+var currClients = make([]*Client, 3) // A singleton per connection type (for benchmarking)
+var currClient *Client               // a singleton so Readings can be de-referenced
```

Comment:
Group all vars under `var(...)`

Reply:
Got rid of the multiple connection types.